ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/PullQueryConsistencyFunctionalTest.java

```diff
@@ -0,0 +1,466 @@
+    makeAdminRequest(
+      REST_APP_0,
+      "CREATE STREAM " + USERS_STREAM
+        + " (" + USER_PROVIDER.ksqlSchemaString(false) + ")"
+        + " WITH ("
+        + "   kafka_topic='" + topic + "', "
+        + "   value_format='JSON');",
+      USER_CREDS
+    );
+    //Create table
+    output = KsqlIdentifierTestUtil.uniqueIdentifierName();
+    sqlTableScan = "SELECT * FROM " + output + ";";
+    List<KsqlEntity> res = makeAdminRequestWithResponse(
+      REST_APP_0,
+      "CREATE TABLE " + output + " AS"
+        + " SELECT " + USER_PROVIDER.key() + ", COUNT(1) AS COUNT FROM " + USERS_STREAM
+        + " GROUP BY " + USER_PROVIDER.key() + ";",
+      USER_CREDS
+    );
+    queryId = extractQueryId(res.get(0).toString());
+    queryId = queryId.substring(0, queryId.length() - 1);
+    waitForTableRows();
+
+    waitForStreamsMetadataToInitialize(
+      REST_APP_0, ImmutableList.of(HOST0, HOST1, HOST2), USER_CREDS);
+  }
+
+  @After
+  public void cleanUp() {
+    REST_APP_0.closePersistentQueries(USER_CREDS);
+    REST_APP_0.dropSourcesExcept(USER_CREDS);
+    APP_SHUTOFFS_0.reset();
+    APP_SHUTOFFS_1.reset();
+    APP_SHUTOFFS_2.reset();
+  }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void shouldPassConsistencyTokenInFanOutTableScan()
+    throws Exception {
+    // Given:
+    ClusterFormation clusterFormation = findClusterFormation(TEST_APP_0, TEST_APP_1, TEST_APP_2);
+    waitForClusterToBeDiscovered(clusterFormation.router.getApp(), 3, USER_CREDS);
+    waitForRemoteServerToChangeStatus(clusterFormation.router.getApp(),
+      clusterFormation.router.getHost(), HighAvailabilityTestUtil.lagsReported(3), USER_CREDS);
+    waitForRemoteServerToChangeStatus(
+      clusterFormation.router.getApp(),
+      clusterFormation.active.getHost(),
+      HighAvailabilityTestUtil::remoteServerIsUp, USER_CREDS);
+    waitForRemoteServerToChangeStatus(
+      clusterFormation.router.getApp(),
+      clusterFormation.standBy.getHost(),
+      HighAvailabilityTestUtil::remoteServerIsUp, USER_CREDS);
+
+    // When:
+    final KsqlRestClient restClient = clusterFormation.router.getApp().buildKsqlClient(USER_CREDS);
+    restClient.setProperty(KSQL_QUERY_PULL_CONSISTENCY_OFFSET_VECTOR_ENABLED, true);
+    ConsistencyOffsetVector offsetVector = new ConsistencyOffsetVector();
+    offsetVector.update("bla", 0, 0);
+    final RestResponse<List<StreamedRow>> res = restClient.makeQueryRequest(sqlTableScan, 1L, null, ImmutableMap.of(KsqlRequestConfig.KSQL_REQUEST_QUERY_PULL_CONSISTENCY_OFFSET_VECTOR, ""));
+    List<StreamedRow> rows = res.getResponse();
+    List<List<?>> values = rows.stream()
+      .skip(HEADER)
+      .filter(sr -> sr.getRow().isPresent())
+      .map(sr -> {
+        return sr.getRow().get().getColumns();
+      })
+      .collect(Collectors.toList());
+
+    // Then:
+    assertThat(rows, hasSize(HEADER + 6));
+    assertThat(rows.get(HEADER+5).getConsistencyToken(), is(not(Optional.empty())));
```

Comment:
What you actually want to check is that the consistency token contains the information you expect i.e., it did not get garbled after all the fan out and response.